### DISCUSSION TITLE
DataTable selectionMode conditional Checkbox

### DISCUSSION THREAD

**Discussion by alqa-dash** [primefaces/community#693] [PrimeVue] [answered]
Hello,

Sometimes it is necessary to mark multiple rows in DataTable with checkboxes (selectionMode="multiple") but excluding some of rows by row property (slotProps.data.someProp) with possibility to check all of them with checkbox in table header (excluding unnecessary of course). 

Unfortunately if <template #body="slotProps"></template> added to <Column selectionMode="multiple"></Column> - checkboxes disappeared.

May be I missed something and it is pretty easy? Any help appreciated.

Thanks!

**Comment by mertsincan** [comment, marked as answer]
Hello,

I think this article can help to you; https://dev.to/mertsincan/handling-big-data-on-datatable-with-checkbox-selection-multiselect-4hgc

Best Regards,

**Comment by alqa-dash** [comment]
Thank you very much! Very useful article. Especially `const isSelectable = (data) => !/o/gi.test(data.name);` from one of examples - never saw this before. 

I tried selectAll and select-all-changed - will use it when needed. 

Unfortunately with "isSelectable" I can disable whole row but not the "<Column selectionMode="multiple"></Column>" which in my case is needed. I'm gonna try to find the way to do it.
